### PR TITLE
Manage parameters in constructor #820

### PR DESCRIPTION
**Pull request by yann300**
Here is the behavior:

If the constructor does not contains parameters:
- Press F5 and it start debugging directly (using default values).
- Create a new state and the contract creation transaction is automatically created.

If the constructor contains parameters:
 - Press F5, a transaction modal dialog appear. This is a light version of the transaction modal dialog, it only contains the list of parameters (balance, value, gas, gasPrice are hidden => use of default values).
 - Create a new state and the contract creation transaction is automatically created (users can change gas, gasPrice, etc...).

Comments are welcome!

### CODE REVIEW

File: mix/qml/TransactionDialog.qml [hunk 1]
```diff
@@ -67,112 +87,159 @@ Window {
...
 		}
+		else
+		{
+			item = TransactionHelper.defaultTransaction();
+			item.functionId = transactionDialog.functionId;
+			item.executeConstructor = isConstructorTransaction;
+		}
+
+		if (isConstructorTransaction)
+			item.functionId = qsTr("Constructor");
```
Comment (arkpar) [inline mix/qml/TransactionDialog.qml]: This should include some special symbol, to avoid potential conflict with function name

Comment (arkpar) [inline mix/qml/TransactionDialog.qml]: Ok, I see it is not used to id the constructor

File: mix/ClientModel.cpp [hunk 2]
```diff
@@ -215,14 +217,20 @@ void ClientModel::showDebugError(QString const& _error)
 	m_context->displayMessageDialog(tr("Debugger"), _error);
 }
 
-ExecutionResult ClientModel::deployContract(bytes const& _code)
+ExecutionResult ClientModel::deployContract(bytes const& _code, TransactionSettings const& ctrTransaction)
```
Comment (gavofyork) [inline mix/ClientModel.cpp]: `_ctrTransaction` (note underscore)

File: mix/ClientModel.h [hunk 3]
```diff
@@ -116,8 +121,8 @@ private slots:
 
 private:
 	QString contractAddress() const;
-	void executeSequence(std::vector<TransactionSettings> const& _sequence, u256 _balance);
-	ExecutionResult deployContract(bytes const& _code);
+	void executeSequence(std::vector<TransactionSettings> const& _sequence, u256 _balance, TransactionSettings const& ctrTransaction = TransactionSettings());
```
Comment (gavofyork) [inline mix/ClientModel.h]: don't kill the underscore!

**Comment by yann300** [comment]
When the user press F5 and if a contract constructor contains parameters, the user is asked to fill the parameters. When a new state is created, the transaction deploying the contract is added in the list and the user can fill in the parameters.

**Pull request merged**